docs(migration-guide): added migration guide for Icon and IconButton (#1483)
contentful/forma-36: MIGRATION.md

When using NPM

> ```bash

Reviewer: maybe we need to do `static=true` here?

Reviewer 2: Not really, this is only used on the github repo, not on the website

Author: my spellchecker was complaining about code block without a type, I'll update it to sh

</Button>;
```

## Icon

Icon component in v4 has a simplified API. Simply import a required icon from `@contentful/f36-icons`.

```tsx static=true
import { ChevronLeftIcon } from '@contentful/f36-icons';

<ChevronLeftIcon size="small" variant="primary" />;
```

> In order to use an iicon from a third-party library, simply pass it as `as` prop:

Reviewer: ```suggestion
In order to use an icon from a third-party library, simply pass it as `as` prop:
```

```tsx static=true
import { Icon } from '@contentful/f36-icons';
import { MdAccessAlarm } from 'react-icons/md';

const AccessAlarm = (props) => {
  return <Icon {...props} as={MdAccessAlarm} variant="secondary" />;
};
```

> Or a custom icon can be rendered as a child with standart html tags:

Reviewer: ```suggestion
There is an option to render custom icon as a child with standard HTML tags:
```
